site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -0,0 +1,248 @@
+from babel_edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+
+
+class RDMEntry:
+    """Building of CDS-RDM entry record."""
+
+    def __init__(self, inspire_metadata):
+        """Initializes the RDM entry."""
+        self.inspire_metadata = inspire_metadata
+        self.errors = []
+
+    def _metadata(self, inspire2cds):
+        """Transformation of metadata."""
+        metadata, errors = inspire2cds.transform_metadata()
+        self.errors.extend(errors)
+        return metadata
+
+    def _files(self, inspire2cds):
+        """Transformation of files."""
+        files, errors = inspire2cds.transform_files()
+        self.errors.extend(errors)
+        return files
+
+    def _parent(self):
+        """Record parent minimal values."""
+        return {
+            "access": {
+                "owned_by": {
+                    "user": 2,  # temporary solution before we have System user
+                }
+            }
+        }
+
+    def _access(self):
+        """Record access minimal values."""
+        return {
+            "record": "public",
+            "files": "public",
+        }
+
+    def build(self):
+        """BPerform building of CDS-RDM entry record."""
+        inspire2cds = Inspire2RDM(self.inspire_metadata)
+        rdm_record = {
+            "metadata": self._metadata(inspire2cds),
+            "files": self._files(inspire2cds),
+            "parent": self._parent(),
+            "access": self._access(),
+        }
+
+        return rdm_record, self.errors
+
+
+class Inspire2RDM:
+    """INSPIRE to CDS-RDM record mapping."""
+
+    def __init__(self, inspire_metadata):
+        """Initializes the Inspire2RDM class."""
+        self.inspire_metadata = inspire_metadata
+        self.metadata_errors = []
+        self.files_errors = []
+
+    def _transform_titles(self):
+        """Mapping of INSPIRE titles to metadata.title and additional_titles."""
+        inspire_titles = self.inspire_metadata.get("titles")
+        rdm_title = None
+        rdm_additional_titles = []
+
+        for i, inspire_title in enumerate(inspire_titles):
+            try:
+                if i == 0:
+                    rdm_title = inspire_title.get("title")
+                else:
+                    rdm_additional_titles.append(
+                        {
+                            "title": inspire_title.get("title"),
+                            "type": {
+                                "id": "alternative-title",
+                            },
+                        }
+                    )
+
+                if inspire_title.get("subtitle"):
+                    rdm_additional_titles.append(
+                        {
+                            "title": inspire_title.get("subtitle"),
+                            "type": {
+                                "id": "subtitle",
+                            },
+                        }
+                    )
+            except Exception as e:
```

Comment:
what kind of exceptions we might get here?

Reply:
We don't know for sure what data will be coming from INSPIRE. For example, could be that `inspire_title.get("title")`  returns `None`. To me it's better if we will know about this